code/controller/controller.ino

```diff
@@ -62,80 +64,71 @@ void setup() {
 }
 
 void loop() {
-  bool is_pressed = false;
+  // bool is_pressed = false;  // unused variable?
+  static unsigned long now=0,next_update=0;
+  now=micros();
+  if (next_update < now+now+REFRESH_INTERVAL) { // time to poll our buttons again!
+	next_update=now+REFRESH_INTERVAL;  // no matter how long it takes us to do all the crap here, our next poll will be ASAP after the refresh time expires.
+	for (int c = 0; c < NUM_COLS; ++c) {
+		byte col = COL_PINS[c];
+		pinMode(col, OUTPUT);
+		digitalWrite(col, LOW);
+		//delayMicroseconds(0);  // perhaps needed to wait for the pin to settle.
+
+		for (int r = 0; r < NUM_ROWS; ++r) {
+		  byte row = ROW_PINS[r];
+		  byte current_button = c * NUM_COLS + r + 1;
+		  
+		  pinMode(row, INPUT_PULLUP);
+  		  //delayMicroseconds(0);  // perhaps needed to wait for the pin to settle.
+		  bool button_state = !digitalRead(row);
+
+		  pinMode(row, INPUT);    // disable the pullup resistor
```

Comment:
I think we can remove this line and the two lines after it for simplicity. This was added because I had connected two of these nodes to the gates of some transistors, but they aren't connected anymore.